javascript/javascript2/week2/hyfBay/main.js

```diff
@@ -0,0 +1,92 @@
+console.log('Script loaded');
+
+const products = getAvailableProducts();
+const productsUl = document.getElementById('productsUl');
+
+console.log(products);
+
+function renderProducts(products) {
+  for (const product of products) {
+    const productLi = document.createElement('li');
+    const productName = document.createElement('h3');
+    const productPrice = document.createElement('p');
+    const productRating = document.createElement('p');
+
+    productsUl.appendChild(productLi);
+    productLi.appendChild(productName);
+    productLi.appendChild(productPrice);
+    productLi.appendChild(productRating);
+
+    productName.innerHTML = product.name;
+    productPrice.innerHTML = `Price: ${product.price}`;
+    productRating.innerHTML = `Rating: ${product.rating}`;
+  }
+}
+
+renderProducts(products);
+
+const searchByName = document.getElementById('search-by-name');
+
+searchByName.addEventListener('input', function () {
+  const searchByNameValue = searchByName.value.toLowerCase();
+  const filteredByName = products.filter((product) => {
+    return product.name.toLowerCase().includes(searchByNameValue);
+  });
+  console.log(filteredByName);
+  productsUl.innerHTML = '';
+  searchByNameValue.length > 0
+    ? renderProducts(filteredByName)
+    : renderProducts(products);
+});
+
+const searchByPrice = document.getElementById('search-by-price');
+
+searchByPrice.addEventListener('change', function () {
+  const searchByPriceValue = searchByPrice.value;
+  const filteredByPrice = products.filter((oneProduct) => {
+    return oneProduct.price <= searchByPriceValue;
+  });
+  productsUl.innerHTML = '';
+  searchByPriceValue > 0
+    ? renderProducts(filteredByPrice)
+    : renderProducts(products);
+});
+
+const selector = document.getElementById('selector');
+
+selector.addEventListener('change', () => {
+  switch (selector.value) {
+    case 'nameAZ':
+      const sortedNameAZ = products.sort((a, b) => (a.name > b.name ? 1 : -1));
```

Comment:
right now capital words come first, then words are started in lower case. if you want to fix this, you can change your code to 
```products.sort((a, b) => (a.name.localeCompare(b.name) > b.name.localeCompare(a.name) ? 1 : -1));```